File: 989 - Add to Array-Form of Integer.py
class Solution:
    def addToArrayForm(self, num: list[int], k: int) -> list[int]:
        res = []
        n = len(num)
        for i in range(n - 1, -1, -1):
            s = num[i] + k % 10
            k //= 10
            if s >= 10:
                s -= 10
                k += 1
            res.append(s)
        while k > 0:
            res.append(k % 10)
            k //= 10
        res = list(reversed(res))
        return res

File: 989 - Add to Array-Form of Integer.js
/**
 * @param {number[]} num
 * @param {number} k
 * @return {number[]}
 */
var addToArrayForm = function (num, k) {
    let res = []
    for (let i = num.length - 1; i >= 0; i--) {
        let sum = num[i] + k % 10;
        k = Math.floor(k / 10);
        if (sum >= 10) {
            sum -= 10;
            k++;
        }
        res.unshift(sum);
    }
    while (k > 0) {
        res.unshift(k % 10);
        k = Math.floor(k / 10);
    }
    return res;
};